VoodooPS2Controller/VoodooPS2Controller.cpp

```diff
@@ -445,6 +456,34 @@ void ApplePS2Controller::resetController(void)
 
 // -- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
+bool ApplePS2Controller::hasMux()
+{
+    writeCommandPort(kCP_WriteMouseOutputBuffer);
+    writeDataPort(0xf0);
+    if (readDataPort(kPS2AuxIdx) != 0xf0)
+        return false;
+    writeCommandPort(kCP_WriteMouseOutputBuffer);
+    writeDataPort(0x56);
+    if (readDataPort(kPS2AuxIdx) != 0x56)
```

Comment:
Can we at least try to name these magic constants?

Reply:
The constants aren't labeled in the Linux source, and I haven't found any other documentation. I atleast put them in one place though.